### ISSUE TITLE
AbstractThrowableAssert with hasMessageMatching? (regular expression matching)

### ISSUE THREAD

**Issue by virgo47** [joel-costigliola/assertj-core#457]
Is it possible to add a method matching the exception message with regular expression? Something like:
```
assertThatThrownBy(() -> xy()))
	.isInstanceOf(MyUberException.class)
	.hasMessageMatching("Not possible to set .* for Object with id .*");
```
This would make easier to migrate from JUnit/TestNG exception message matching, but it is also much more flexible on its own.

**Comment by joel-costigliola** [comment]
it is ! :)

**Comment by virgo47** [comment]
I'm not sure I understand but that's probably because of my original phrasing. :-) Do you mean I can implement my own ``AbstractThrowableAssert`` implementation? It is of course possible.

I thought it would be nice to have it out of the box. If it's not planned at all, you may close the issue. Or can I try to implement it as a pull request?

**Comment by joel-costigliola** [comment]
No, I was saying that we would add `hasMessageMatching` :)

**Comment by joel-costigliola** [comment]
addressed by #446
